pkgs/development/tools/misc/blackmagic/default.nix

```diff
@@ -1,36 +1,39 @@
 { stdenv, lib, fetchFromGitHub
-, gcc-arm-embedded, binutils-arm-embedded, libftdi
+, gcc-arm-embedded, libftdi1
 , python, pythonPackages
 }:
 
 with lib;
 
 stdenv.mkDerivation rec {
   name = "blackmagic-${version}";
-  version = "1.6.1";
+  version = "2019-07-28";
```

Comment:
A minor version nit:

Normally, I try to keep the latest stable version number included in the `version` string, as well as the number of commits since that tag, and the git hash, rather than using the date. Nix will look at `version` in order to determine if e.g. `nix-env -u` needs to upgrade a package. (Check out `builtins.compareVersions` for more.)

The format I use is described by [the liburing expression](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/libraries/liburing/default.nix), basically (this same convention is used for `nixUnstable`, as well.) This is mostly to make the versioning a little more robust in case you need to, say, do two updates in one day (because someone reports a regression, or upstream fixes something important you report after pushing, etc. This is rare, though)

It might be better, in this case, though, to match the convention upstream sets in their makefiles with `git describe`. Yet this includes the `v` character, which is inappropriate. So, I would suggest something like this, then:

```nix
stdenv.mkDerivation rec {
  name = "blackmagic-${version}";
  # version is given by `git describe --always`.
  # note: do not include starting letter 'v' in the string!
  version = "1.6.1-311-gfbf1963";

  ...

  postPatch = ''
    --replace '$(shell git describe --always --dirty)' 'v${version}'
  '';
}
```

That is: just eliminate `firmwareVersion`, and put the 'v' character in the `postPatch` phase.

I think this makes things a little nicer but I wouldn't hold things up on it. (I don't know how others feel about e.g. dates as commits, but I know we have a lot of every-possible-thing-you-can-imagine laying around...)

Reply:
This is how I originally had things but @worldofpeace asked me to change it to use a date :) I'm happy to put it back if that's desired though.